Tests/SentryTests/Integrations/SentryCrash/SentryCrashScopeObserverTests.swift

```diff
@@ -281,6 +281,17 @@ class SentryCrashScopeObserverTests: XCTestCase {
         assertEmptyScope()
     }
     
+    func testModifyScopeFromDifferentThreads() {
+        let scope = Scope()
+        scope.add(SentryCrashScopeObserver(maxBreadcrumbs: 100))
+        
+        testConcurrentModifications(asyncWorkItems: 20, writeLoopCount: 100, writeWork: { i in
+            let user = User()
+            user.name = "name \(i)"
+            scope.setUser(user)
+        })
```

Comment:
We should assert that all of the writes are recorded in the order we expect. I see some other usages of `testConcurrentModifications` making assertions in its `readWork` block.

Reply:
Looked at `testConcurrentModifications` implementation, and the DispatchGroups do not enforce execution order among the concurrent tasks. I don't see how we could assert write order here?

Reply:
resolving this so we can merge, we can revisit later if this comes up again.